tests/elements.rs

```diff
@@ -0,0 +1,193 @@
+//! Element tests
+#[macro_use]
+extern crate serial_test;
+extern crate fantoccini;
+extern crate futures_util;
+
+use crate::common::sample_page_url;
+use fantoccini::keys::Key;
+use fantoccini::{error, Client, Locator};
+
+mod common;
+
+async fn element_is(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+    let mut elem = c.find(Locator::Id("checkbox-option-1")).await?;
+    assert!(elem.is_enabled().await?);
+    assert!(elem.is_displayed().await?);
+    assert!(!elem.is_selected().await?);
+    elem.click().await?;
+    let mut elem = c.find(Locator::Id("checkbox-option-1")).await?;
+    assert!(elem.is_selected().await?);
+
+    assert!(
+        !c.find(Locator::Id("checkbox-disabled"))
+            .await?
+            .is_enabled()
+            .await?
+    );
+    assert!(
+        !c.find(Locator::Id("checkbox-hidden"))
+            .await?
+            .is_displayed()
+            .await?
+    );
+    Ok(())
+}
+
+async fn element_attr(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+    let mut elem = c.find(Locator::Id("checkbox-option-1")).await?;
+    assert_eq!(elem.attr("id").await?.unwrap(), "checkbox-option-1");
+    assert!(elem.attr("invalid-attribute").await?.is_none());
+    Ok(())
+}
+
+async fn element_prop(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+    let mut elem = c.find(Locator::Id("checkbox-option-1")).await?;
+    assert_eq!(elem.prop("id").await?.unwrap(), "checkbox-option-1");
+    assert_eq!(elem.prop("checked").await?.unwrap(), "false");
+    assert!(elem.attr("invalid-property").await?.is_none());
+    Ok(())
+}
+
+async fn element_css_value(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+    let mut elem = c.find(Locator::Id("checkbox-hidden")).await?;
+    assert_eq!(elem.css_value("display").await?, "none");
+    assert_eq!(elem.css_value("invalid-css-value").await?, "");
+    Ok(())
+}
+
+async fn element_tag_name(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+    let mut elem = c.find(Locator::Id("checkbox-option-1")).await?;
+    assert_eq!(elem.tag_name().await?, "input");
```

Comment:
Should we `.to_lower_case()` here, or does the spec guarantee lower-case?

Reply:
I wouldn't expect the spec to guarantee lower-case so I've added that.